storage/storage.go

```diff
@@ -1634,6 +1634,24 @@ func conditionsQuery(gen int64, conds *Conditions) string {
 	return string(buf)
 }
 
+// setConditionsHeaders sets precondition request headers for downloads
+// using the XML API. It assumes that the conditions have been validated.
```

Comment:
Let's maybe move this to reader.go since it's only used there?

Reply:
Done. That makes sense, thanks!